src/tbb/static_string.h

```diff
@@ -0,0 +1,124 @@
+/*
+    Copyright (c) 2022 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#ifndef __TBB_static_string
+#define __TBB_static_string
+
+#include "oneapi/tbb/detail/_config.h"
+
+#include <array>
+#include <cstring>
+#include <algorithm> // for std::min, std::fill
+
+#include "literal_const_string.h"
+#include "string_view.h"
+
+namespace tbb {
+namespace detail {
+//should we increase it here ?
+namespace r1 {
+
+template<std::size_t N>
+class static_string{
+
+    static_assert(N > 0,"no point in creating static_string with no storage allocated");
+    std::array<char, N + 1> ar = {0};
+    std::size_t m_size = 0;
+
+public:
+
+    //While initialization of ar can be omitted due to C++ rules, gcc 4.8 insist on it
+    static_string(std::size_t sz = 0) : ar{0}, m_size (sz) {};
+
+
+    std::size_t size()  const noexcept      { return m_size;}
+    std::size_t max_size() const noexcept   { return ar.max_size() -1;}
+
+    //FIXME: cover with tests
+    void clear() noexcept {
+        m_size = 0;
+        ar[m_size] = 0;
```

Comment:
Is it okay that `m_size` ~~accidentally~~ can be greater than template argument `N`?

The question is related not only to this method.

I think `assert(N >= m_size)` in the default constructor is enough to resolve this issue.

Reply:
maybe fill `ar` with zeros in `clear`?